packages/ra-core/src/auth/useLogoutIfAccessDenied.ts

```diff
@@ -92,12 +96,13 @@ const useLogoutIfAccessDenied = (): LogoutIfAccessDenied => {
                     if (logoutUser) {
                         logout({}, redirectTo);
                     } else {
+                        dispatch(fetchEnd());
```

Comment:
i don't understand why this is necessary, as performXXXQuery already dispatches a FAILURE action with a FETCH_ERROR meta, that the loading reducer should use to decrease the loading state. 

Could you elaborate in a comment?

Also, if a dataProvider action is necessary, I'd prefer it to be dispatched in performXXXQuery rather than in this hook.

Reply:
The problem is that callback always returns `true`. If I make it return `logoutUser`, the user gets logged out when `logoutUser` is `false`

Reply:
you're right. then the change should be done in each of the 3 performXXXQuery scripts instead: if `loggedOut` is true, the method should dispatch a FETCH_END before returning.